var/spack/repos/builtin/packages/rocsolver/package.py

```diff
@@ -44,6 +44,7 @@ class Rocsolver(CMakePackage):
     depends_on('cmake@3.8:', type='build', when='@4.1.0:')
     depends_on('cmake@3.5:', type='build')
     depends_on('fmt@7.1.3:', type='build', when='@4.5.0:')
+    conflicts('rocsolver@4.5.2', when='^fmt@8.1.0:8.1.1')
```

Comment:
I expect there was a breaking api change? Maybe best to just specify compat bounds in depends_on?

```suggestion
    depends_on('fmt@:8', type='build', when='@4.5.0:')
```

Reply:
(i think in general we could just specify upperbounds for any package respecting semver by default since major version bumps are breaking... it will prevent future headaches)

Reply:
I agree. I just fixed this, with the only change being that `fmt@8.0.1` is actually OK.

Reply:
so much for semver ;p